x-pack/libbeat/reader/parquet/parquet_test.go

```diff
@@ -0,0 +1,282 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package parquet
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"testing"
+
+	"github.com/apache/arrow/go/v11/arrow"
+	"github.com/apache/arrow/go/v11/arrow/array"
+	"github.com/apache/arrow/go/v11/arrow/memory"
+	"github.com/apache/arrow/go/v11/parquet/pqarrow"
+	"github.com/stretchr/testify/assert"
+)
+
+// all test files are read from/stored within the "testdata" directory
+const testDataPath = "testdata/"
+
+func TestParquetWithRandomData(t *testing.T) {
+	testCases := []struct {
+		columns int
+		rows    int
+	}{
+		{
+			columns: 10,
+			rows:    20,
+		},
+		{
+			columns: 15,
+			rows:    30,
+		},
+		{
+			columns: 5,
+			rows:    50,
+		},
+		{
+			columns: 10,
+			rows:    1000,
+		},
+		{
+			columns: 15,
+			rows:    1000,
+		},
+		{
+			columns: 15,
+			rows:    10000,
+		},
+	}
+
+	for i, tc := range testCases {
+		name := fmt.Sprintf("Test parquet files with rows=%d, and columns=%d", tc.rows, tc.columns)
+		t.Run(name, func(t *testing.T) {
+			tempDir := t.TempDir()
+			fName := fmt.Sprintf("%s/%s_%d.parquet", tempDir, "test", i)
+			data := createRandomParquet(t, fName, tc.columns, tc.rows)
+			file, err := os.Open(fName)
+			if err != nil {
+				t.Fatalf("Failed to open parquet test file: %v", err)
+			}
+			defer file.Close()
+
+			cfg := &Config{
+				// we set ProcessParallel to true as this always has the best performance
+				ProcessParallel: true,
+				// batch size is set to 1 because we need to compare individual records one by one
+				BatchSize: 1,
+			}
+			rows := readAndValidateParquetFile(t, cfg, file, data)
+			// asserts of number of rows read is the same as the number of rows written
+			assert.Equal(t, rows, tc.rows)
+		})
+	}
+}
+
+// readAndValidateParquetFile reads the parquet file and validates the data
+func readAndValidateParquetFile(t *testing.T, cfg *Config, file *os.File, data map[string]bool) int {
+	sReader, err := NewStreamReader(file, cfg)
+	if err != nil {
+		t.Fatalf("failed to init stream reader: %v", err)
+	}
+
+	rowCount := 0
+	for sReader.Next() {
+		val, err := sReader.Record()
+		if err != nil {
+			t.Fatalf("failed to read stream: %v", err)
+		}
+		if val != nil {
+			rowCount++
+			// this is where we check if the column values are the same as the ones we wrote
+			if _, ok := data[string(val)]; !ok {
```

Comment:
```suggestion
			if !data[string(val)] {
```
`data` is a `map[string]bool`.